src/engine/enginepregain.h

```diff
@@ -31,10 +31,14 @@ class EnginePregain : public EngineObject {
     virtual ~EnginePregain();
 
     void setSpeed(double speed);
+    void setScratching(bool scratching);
```

Comment:
documentation:
// If the user is scratching and the record reverses direction, the volume will be ramped to zero and back up again to mimic a vinyl scratch.  If the user is not scratching and the direction is reversed (e.g. reverse button is pressed), the audio will be immediately reversed without a ramp to zero.